### PR TITLE
Feat: API integration

### PR DESCRIPTION
**Pull request by mdarif6**
- API integration.
- Authentication applied.
- ReadMe updated.
- Code Refactored.

### CODE REVIEW

**Review by ravikumar1002 — APPROVED** [review]
looks good to merge
remove all comment

File: src/pages/archive-page/ArchiveAside.jsx [hunk 1]
```diff
@@ -37,14 +44,26 @@ export default function ArchiveAside() {
...
+                <i className="far fa-user-circle"></i>
+              </div>
+              <Link className="link-style" to="/login">
+                <div>Login</div>
+              </Link>
+            </div>
+          )}
         </div>
-        <div className="join-button">
+        {/* <div className="join-button">
```
Comment (Shah-Arsalan) [inline src/pages/archive-page/ArchiveAside.jsx]: remove this

File: src/pages/archive-page/ArchiveMain.jsx [hunk 2]
```diff
@@ -1,26 +1,111 @@
...
+        `/api/archives/restore/${note._id}`,
+        {},
+        {
+          headers: {
+            authorization: token,
+          },
+        }
+      );
+
+      // if (response.status === 200) {
```
Comment (Shah-Arsalan) [inline src/pages/archive-page/ArchiveMain.jsx]: this also

**Review by Shah-Arsalan — COMMENTED** [review]
Code looks good . The file structure is also pretty well done . You can merge

File: src/pages/archive-page/ArchiveAside.jsx (hunk 1 above)
Comment (mdarif6) [reply src/pages/archive-page/ArchiveAside.jsx]: done

File: src/pages/archive-page/ArchiveMain.jsx (hunk 2 above)
Comment (mdarif6) [reply src/pages/archive-page/ArchiveMain.jsx]: removed

**Pull request merged**